### ISSUE TITLE
hoisting answer is incorrect.

### ISSUE THREAD

**Issue by DelliriumX** [yangshun/front-end-interview-handbook#168]
The examples for an answer is incorrect:
https://github.com/yangshun/front-end-interview-handbook/blob/master/questions/javascript-questions.md#explain-hoisting

Believe it or not, variables declared using `let` or `const` are hoisted, their namespace gets hoisted however they do not get declared until the interpreter reaches the line of code on which the declaration happens. To demonstrate this:

```javascript
var a = "foo";
function proofOfHoisting(){
  console.log(a);
  let a = "bar";
}
```

If hoisting was to not be a thing for `let` and/or `const` then running this function would actually log `foo`, however it does not, it gives a reference error because the inner variable is hoisted, but not declared, however because of hoisting the outer variable is not visible.

**Comment by Redmega** [comment]
user Bergi has an excellent answer on Stack Overflow regarding hoisting of let/const https://stackoverflow.com/a/31222689/2748503

**Comment by DelliriumX** [comment]
Yes that is a very good answer/explenation, however what we have in this repo is specifically saying this is not the case claimin; "however let/const are NOT hoisted". I've had multiple interviews where senior level programmers had asked this and were in the wrong, because people learn it wrong. This should be fixed.

**Comment by yangshun** [comment]
Thanks for the issue, PRs are welcome 😄

**Comment by yangshun** [comment]
Fixed via e96fc61.
